### PR TITLE
[GEN-1847]: add AppDynamics as destination

### PR DESCRIPTION
**Pull request by BenElferink**
This pull request introduces support for AppDynamics as a new managed destination in the Odigos project. The changes include updates to configuration files, documentation, and the addition of new files to support the integration.

Key changes:

### New Destination Support:

* Added `AppDynamics` as a new managed destination type in the configuration files (`common/config/appdynamics.go`, `common/config/root.go`, `common/dests.go`). [[1]](diffhunk://#diff-18211d7ae505c48c4dde4ef3e379edc0c938fab592c7faf2d3b831aff09c1a65R1-R66) [[2]](diffhunk://#diff-35f076c473aa76e7717e17ac33041f73ec6b16e26fe8a1ab50bf5963cb7fcd8eL18-R18) [[3]](diffhunk://#diff-e0ef4d5cecfc896240ae7392424f227db50784f83d0ddc0c317db59066f8757bR6)
* Created a new YAML configuration file for AppDynamics (`destinations/data/appdynamics.yaml`).

### Documentation Updates:

* Updated the `README.md` to include AppDynamics in the list of managed destinations.
* Added a new documentation page for configuring AppDynamics (`docs/backends/appdynamics.mdx`).
* Updated the backends overview and navigation to include AppDynamics (`docs/backends-overview.mdx`, `docs/mint.json`). [[1]](diffhunk://#diff-26115b197a8b9dd8ee351f05b2cade47da5acd788d74b9f962a325d3e1b919a2L2-R41) [[2]](diffhunk://#diff-c91a604899dfef4b2494c317f4fd39a7f22b79986095f580399347293d534debL203-R203) [[3]](diffhunk://#diff-c91a604899dfef4b2494c317f4fd39a7f22b79986095f580399347293d534debL220-R221) [[4]](diffhunk://#diff-c91a604899dfef4b2494c317f4fd39a7f22b79986095f580399347293d534debR231)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -82,44 +82,49 @@ For more details, see our [quickstart guide](https://docs.odigos.io/intro).
...
+| Google Cloud Storage    |   ✅   |         |  ✅  |
+| Grafana Cloud           |   ✅   |   ✅    |  ✅  |
+| Honeycomb               |   ✅   |   ✅    |  ✅  |
+| Last9                   |   ✅   |   ✅    |      |
+| Lightstep               |   ✅   |         |      |
+| Logz.io                 |   ✅   |   ✅    |  ✅  |
+| New Relic               |   ✅   |   ✅    |  ✅  |
+| OpsVerse                |   ✅   |   ✅    |  ✅  |
+| OTLP                    |   ✅   |   ✅    |  ✅  |
+| OTLP HTTP               |   ✅   |   ✅    |  ✅  |
```
Comment (blumamir) [inline README.md:109]: This is not managed

File: README.md [hunk 2]
```diff
@@ -82,44 +82,49 @@ For more details, see our [quickstart guide](https://docs.odigos.io/intro).
...
+| qryn                    |   ✅   |   ✅    |  ✅  |
```
Comment (blumamir) [inline README.md:110]: self hosted

File: common/config/appdynamics.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+	endpoint, exists := config[APPDYNAMICS_ENDPOINT]
+	if !exists {
+		return errors.New("AppDynamics Endpoint (\"APPDYNAMICS_ENDPOINT\") not specified, AppDynamics will not be configured")
+	}
+
+	apiKey, exists := config[APPDYNAMICS_API_KEY]
+	if !exists {
+		return errors.New("AppDynamics API Key (\"APPDYNAMICS_API_KEY\") not specified, AppDynamics will not be configured")
+	}
```
Comment (blumamir) [inline common/config/appdynamics.go:32]: this should be a secret

File: docs/backends/appdynamics.mdx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+2. Navigate to the OpenTelemetry page:
+
+```bash
+https://{YOUR_USERNAME_HERE}.saas.appdynamics.com/controller/#/apm/otel
+```
+
+3. Click on the `Configuration` tab, then click on `exporters`.
+4. Copy your `Endpoint`, and generate an `API Key`.
+
+- **Endpoint** - AppDynamics OTeL Endpoint.
```
Comment (blumamir) [inline docs/backends/appdynamics.mdx:17]: Is this the text in the website?

```suggestion
- **Endpoint** - AppDynamics OpenTelemetry OTLP/http Endpoint.
```

File: docs/backends/appdynamics.mdx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+- **API Key** - AppDynamics OTeL API Key.
```
Comment (blumamir) [inline docs/backends/appdynamics.mdx:18]:
```suggestion
- **API Key** - AppDynamics API Key.
```

**Comment by BenElferink** [comment]
Added processor params, and it works like a charm 😃

**Review by tamirdavid1 — APPROVED** [review]

File: common/config/appdynamics.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+	accountName, exists := config[APPDYNAMICS_ACCOUNT_NAME]
+	if !exists {
+		return errors.New("AppDynamics Account Name (\"APPDYNAMICS_ACCOUNT_NAME\") not specified, AppDynamics will not be configured")
+	}
+
+	endpointParts := strings.Split(endpoint, ".")
+	if len(endpointParts) > 0 {
+		// replace the first part of the endpoint with the account name (instead of collecting another input from the user)
+		endpointParts[0] = accountName
```
Comment (blumamir) [inline common/config/appdynamics.go:62]: Can you describe what exactly you are doing here. what is the first part of the endpoint? can it contain dot as well which will break this logic?

**Review by blumamir — APPROVED** [review]

**Pull request merged**